### PR TITLE
Chunk based supervoxel loading from Chunked Graph

### PR DESCRIPTION
**Pull request by nkemnitz**
Calling the ChunkedGraph's getLeaves() function for visible chunks, only (instead of the whole dataset). Allows displaying huge objects without crashing the graphserver and/or neuroglancer.

TODO: Use bigger chunks when zooming out, only load "large" supervoxels

### CODE REVIEW

File: src/neuroglancer/chunked_graph/backend.ts [hunk 1]
```diff
@@ -14,30 +14,150 @@
...
+  constructor(rpc: RPC, options: any) {
+    super(rpc, options);
     this.graphurl = options['url'];
-    super.initializeSharedObject(rpc, options['id']);
+    this.rootSegments = <Uint64Set>rpc.get(options['rootSegments']);
+    this.visibleSegments3D = <Uint64Set>rpc.get(options['visibleSegments3D']);
+    this.segmentEquivalences = <SharedDisjointUint64Sets>rpc.get(options['segmentEquivalences']);
+
+    this.sources = new Array<ChunkedGraphChunkSource[]>();
+    for (const alternativeIds of options['sources']) {
```
[the remark this answers is not in the document]
Comment (nkemnitz) [reply src/neuroglancer/chunked_graph/backend.ts]: You could probably use it like that, but for now we only have a single scale and a single source.

File: src/neuroglancer/gpu_hash/hash_table.ts [hunk 2]
```diff
@@ -472,29 +480,26 @@ export class HashMapUint64 extends HashTableBase {
...
    */
   [Symbol.iterator]() {
     return this.entries();
   }
 
   /**
-   * Iterates over entries.  The same temporary value will be modified and yielded at every
-   * iteration.
+   * Iterates over entries.
+   * Creates new Uint64 objects at every iteration (otherwise spread and Array.from() fail)
```
[the remark this answers is not in the document]
Comment (nkemnitz) [reply src/neuroglancer/gpu_hash/hash_table.ts]: Hard to say. Before we used the graph server, our hash sets barely contained more than 100 elements. Now we are showing 100,000+, which puts more stress on everything.

File: src/neuroglancer/chunked_graph/backend.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (william-silversmith) [reply src/neuroglancer/chunked_graph/backend.ts]: note: scale = resolution

[the remark this answers is not in the document]
Comment (william-silversmith) [reply src/neuroglancer/chunked_graph/backend.ts]: This is only due to how Jeremy expects the format of this output.

**Pull request merged**